bodo/pandas/_executor.h

```diff
@@ -23,19 +23,25 @@ class Executor {
         PhysicalPlanBuilder builder;
         builder.Visit(*plan);
         pipelines = std::move(builder.finished_pipelines);
-
-        if (builder.active_pipeline != nullptr) {
-            pipelines.push_back(builder.active_pipeline->BuildEnd(out_schema));
-        }
+        assert(builder.active_pipelines.size() == 1);
```

Comment:
What does multiple active pipelines mean? Why is it necessary? There is only one pipeline that is being built at a time. Operators like LIMIT will just finish a pipeline and start a new active pipeline. We are following DuckDB's code and I think we should keep doing it.

Reply:
Yes, limit doesn't need multiple active pipelines but joins do.  It is a post-order traversal of the graph and you start a pipeline at one leaf that carries through to the root of the tree but when you get to a node with more than one child then one of those children will be in the main pipeline but the other has to form its own pipeline.  The original pipeline say with the left sub-tree is still active in the sense that you will add more to it eventually but while you are traversing the right sub-tree you will have a different active pipeline (top of the stack).

Reply:
To be clear, this is talking about pipeline formation, not pipeline execution.  The loop that executes pipelines does go one at a time.